live/client.py

```diff
@@ -132,3 +134,52 @@ def stop_event(self, event_id):
         # Send request and do exception handling
         self.send_request(http_method="POST", url=url,
                           headers=headers, body=body)
+
+    def find_devices_in_use(self):
+        events_url = f'{self.server_ip}/live_events'
+        events_headers = self.generate_headers(events_url)
+        events_xml = self.send_request(
+            http_method="GET", url=events_url, headers=events_headers)
+
+        events_list = ET.fromstring(events_xml.text)
+
+        # Find in use devices from all events
+        in_use_devices = set()
+        for event in events_list:
+            status = event.find('status').text
+            for event_input in event.findall('input'):
+                device_input = event_input.find('device_input')
+                if device_input:
+                    device_name = device_input.find('device_name')
+                    if status in ('preprocessing', 'running'):
+                        in_use_devices.add(device_name.text)
+
+        return in_use_devices
+
+    def get_input_devices(self):
+        devices_url = f'{self.server_ip}/devices'
+        devices_headers = self.generate_headers(devices_url)
+        devices_xml = self.send_request(
+            http_method="GET", url=devices_url, headers=devices_headers)
+
+        devices_list = ET.fromstring(devices_xml.text)
+
+        # Find all devices info
+        all_devices = \
+            {device.find('device_name').text for device in devices_list}
+
+        devices_info_dict = xmltodict.parse(devices_xml.text)[
```

Comment:
the `_dict` in the end of  this variable is an anti-pattern: https://docs.quantifiedcode.com/python-anti-patterns/readability/putting_type_information_in_a_variable_name.html